web3/types.py

```diff
@@ -149,8 +149,8 @@ class FormattersDict(TypedDict, total=False):
 class FilterParams(TypedDict, total=False):
     address: Union[Address, ChecksumAddress, List[Address], List[ChecksumAddress]]
     blockHash: HexBytes
-    from_block: BlockIdentifier
-    to_block: BlockIdentifier
+    fromBlock: BlockIdentifier
```

Comment:
I think ultimately we make these changes here to camel case and everything else that we can leave snake cased should probably be snake cased. My reasoning is that the method arguments to python functions are expected to be snake cased, by convention. They are just variables up until we construct this `TypedDict`. Once we start building the `TypedDict`, this is the one translation step that basically says "Ok, we passed everything around using python conventions but now we need to speak to a service that does not use python conventions". The way I see it behaving is these `TypedDict` request param classes are the translation layer between python and provider and I think ultimately this is the only place we need the explicit change to camelCase.

I'm curious though if by doing this we run into complications elsewhere. Thoughts @linda-le1 @kclowes @pacrob?

Reply:
Yeah, I think I'm in the same boat. Basically anything that goes to/from the node should be camelCase, and everything else should be snaked.